src/components/Nav/index.tsx

```diff
@@ -102,83 +63,26 @@ const Nav = () => {
-                    variant="ghost"
-                    isSecondary
-                    px={{ base: "2", xl: "3" }}
-                    _hover={{
-                      transform: "rotate(10deg)",
-                      color: "primary.hover",
-                    }}
-                    onClick={toggleColorMode}
-                  />
-
-                  {/* Locale-picker menu */}
-                  <LanguagePicker
-                    placement="bottom-end"
-                    minH="unset"
-                    maxH="75vh"
-                    w="xs"
-                    inset="unset"
-                    top="unset"
-                    menuState={languagePickerState}
-                  >
-                    <MenuButton
-                      as={Button}
-                      name={DESKTOP_LANGUAGE_BUTTON_NAME}
-                      ref={languagePickerRef}
-                      variant="ghost"
-                      color="body.base"
-                      transition="color 0.2s"
-                      px={{ base: "2", xl: "3" }}
-                      _hover={{
-                        color: "primary.hover",
-                        "& svg": {
-                          transform: "rotate(10deg)",
-                          transition: "transform 0.5s",
-                        },
-                      }}
-                      _active={{
-                        color: "primary.hover",
-                        bg: "primary.lowContrast",
-                      }}
-                      sx={{
-                        "& svg": {
-                          transform: "rotate(0deg)",
-                          transition: "transform 0.5s",
-                        },
-                      }}
-                    >
-                      <Icon
-                        as={BsTranslate}
-                        fontSize="2xl"
-                        verticalAlign="middle"
-                        me={2}
-                      />
-                      <Text hideBelow="lg" as="span">
-                        {t("common:languages")}&nbsp;
-                      </Text>
-                      {locale!.toUpperCase()}
-                    </MenuButton>
-                  </LanguagePicker>
-                </HStack>
-              )}
+              {/* avoid rendering desktop menu version on mobile */}
+              <Show above="md">
+                <Search {...searchModalDisclosure} />
+                <DesktopNavMenu toggleColorMode={toggleColorMode} />
+              </Show>
 
-              <MobileNavMenu
-                {...mobileNavProps}
-                linkSections={linkSections}
-                hideFrom="md"
-                toggleSearch={searchModalDisclosure.onOpen}
-                drawerContainerRef={navWrapperRef}
-              />
+              <Show below="md">
+                {/* Mobile */}
+                {/* use Suspense to display the Search & the Menu at the same time */}
```

Comment:
> use Suspense to display the Search & the Menu at the same time

I see we have `<Search {...searchModalDisclosure} />` declared both in the Desktop suspense and Mobile suspense... I take it this is why? So items render together at the same time in each environment?

Reply:
Exactly. We could move it out of the suspense wrapper to make it simpler, but we will have this delayed experience, showing the search before the hamburger icon, which is what we want to avoid.